index.bs

```diff
@@ -10183,44 +10226,25 @@ Constructors</h5>
 		the following steps are performed on the <a>rendering thread</a>.
 
 		<div algorithm="AudioWorkletProcessor()">
-			This constructor is invoked by processing a control
-			message queued by the {{AudioWorkletNode()|constructor}}
-			of the associated {{AudioWorkletNode}}. This control
-			message carries the [=processor construction data=] that
-			consists of:
-			<var>nodeName</var>,
-			<var>serializedProcessorPort</var>,
-			<var>serializedOptions</var>,
-			and <var>node</var>.
-
 			If any of these steps throws an exception, abort the rest of
 			steps and <a>queue a task</a> to fire an
 			<a href="https://www.w3.org/TR/html50/webappapis.html#the-errorevent-interface">ErrorEvent</a>
 			named <code>processorerror</code> at the {{AudioWorkletNode}}
 			on the <a>control thread</a> with the relevant information about
 			the error.
 
-			1. If there is no [=processor construction data=]
-				passed from the associated {{AudioWorkletNode}},
-				throw a {{TypeError}}.
-
-			1. Let <var>processorPort</var> be
-				[$StructuredDeserializeWithTransfer$](<var>serializedProcessorPort</var>,
-				the current Realm).
+			1. If there is no provided [=nodePort=] at the
```

Comment:
This doesn't seem to be well-defined, unfortunately.  In particular, what does "invocation of the constructor" mean?  Keep in mind that we can have a subclass of `AudioWorkletProcessor` that invokes multiple `AudioWorkletProcessor` constructors, via `super()` or not.  And none of those invocations can provide a nodePort, note.

This part really needs to have a clear processing model spelled out.  Where is the port information coming from, and does the act of looking it up remove it from that place or not?

Reply:
The port is a part of the processor construction data. It is not an explicit argument from the constructor.

> Keep in mind that we can have a subclass of AudioWorkletProcessor that invokes multiple AudioWorkletProcessor constructors, via super() or not. And none of those invocations can provide a nodePort, note.

I think that's the intention? The AWP constructor cannot be invoked by the user code. So if there is no provided `port` object, the construction can't succeed. I think this is also closely related to how the processor construction data is passed and used. I'll take a look at the infra spec.

Reply:
Your intention is that `super()` (which is an invocation of the AWP constructor in user code!) should be considered as having a port provided based on something about how the original subclass constructor was called, right?  I just think this part should be spelled out so it can be implemented interoperably.

Reply:
> I just think this part should be spelled out so it can be implemented interoperably.

I agree but can we deal with that in a follow-up PR?

Reply:
We can if you would like, but dealing with it was the primary purpose of _this_ PR....  I feel like we're just going around in circles without getting closer to this key part actually being specified, and I am unfortunately running out of the time I have allocated to this.

Reply:
Though I guess if the goal here is rather to narrowly fix https://github.com/WebAudio/web-audio-api/issues/2021 then fine.  But we do still need to fix the port issue...

Reply:
Yes. I think it's better to create a new PR that specifically addresses: the AudioWorkletProcessor's construction with the processor construction data.

Thanks for your time and effort on this. It's been a great help.